SConstruct

```diff
@@ -415,6 +415,27 @@ if windows:
 env = Environment(**envDict)
 del envDict
 
+# The Scons 'default' tool enables a lot of tools that we don't actually need to enable.
+# On platforms like Solaris, it actually does the wrong thing by enabling the sunstudio
+# toolchain first. As such it is simpler and more efficient to manually load the precise
+# set of tools we need for each platform.
+# If we aren't on a platform where we know the minimal set of tools, we fallback to loading
+# the 'default' tool.
+def decide_platform_toolchain():
+    if windows:
+        # we only support MS toolchain on windows
+        return ['msvc', 'mslink', 'mslib']
+    elif linux:
+        return ['gcc', 'g++', 'gnulink', 'ar']
+    elif solaris:
+        return ['gcc', 'g++', 'gnulink', 'sunar']
```

Comment:
I think we just want 'ar' here for sun, as sunar affects SHLINKFLAGS and adds -G